core/src/main/java/com/linecorp/armeria/client/retry/RetryConfig.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.retry;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.common.Response;
+
+/**
+ * Holds retry config used by a {@link RetryingClient}.
+ * A {@link RetryConfig} instance encapsulates the used {@link RetryRule}, maxTotalAttempts,
+ *  * and responseTimeoutMillisForEachAttempt.
+ */
+public final class RetryConfig<T extends Response> {
+    private final int maxTotalAttempts;
+    private final long responseTimeoutMillisForEachAttempt;
+    @Nullable private final RetryRule retryRule;
+    @Nullable private final RetryRuleWithContent<T> retryRuleWithContent;
+    @Nullable private final RetryRule fromRetryRuleWithContent;
+    private final int maxContentLength;
+    private final boolean needsContentInRule;
+
+    /**
+     * Returns a new {@link RetryConfigBuilder} with the default values from Flags.
+     * Uses a {@link RetryRule}.
+     */
+    public static <T extends Response> RetryConfigBuilder<T> builder(RetryRule retryRule) {
```

Comment:
Please move all static factory methods before the member fields.

Reply:
Done